### PR TITLE
Diqing/dev/bc/mongodb atlas integration

### PR DESCRIPTION
**Pull request by changdiqing**
Created MongoDB development database (not for production deployment!)
Created MongoDB client at backend
Added jwt support
Added hashing password support (bcrypt)

### CODE REVIEW

**Comment by Apollo1840** [comment]
@changdiqing are you sure that once this is merged, **everyone** could seamlessly keep their development? in another word, run those 2 servers locally will not cause any trouble.

File: backend/studytimeboard/routes.py [hunk 1]
```diff
@@ -23,11 +23,32 @@
...
 from .constant import *
 from .app_utils import *
 
+# database proxies
+from .utils.mongodb_client import users_collection
+# utils for hasing password and jwt encode/decode
+from .utils.bcrypt_utils import hash_password, validate_password
+from .utils.jwt_utils import jwt_encode, jwt_decode_identity, jwt_verify
+
+def login_required():
```
Comment (Apollo1840) [inline backend/studytimeboard/routes.py:39]: please move this to app_utils.py.

app_utils is a place to put all not-route function. (will be refactored, but so far please put it into the app_utils)

**Review by Apollo1840 — COMMENTED** [review]
reviewed, fantastic work!

I want to refactor jwt and bcrypt into one class called securityService(or secretService) with .decode_identity_from_request(), encode_identity_to_token(), .validate_password(), .hash_password() , so that developer would not necessary to know how security system works to develop and use the security system. But probably I will do this later.

**Comment by changdiqing** [comment]
> reviewed, fantastic work!
> 
> I want to refactor jwt and bcrypt into one class called securityService(or secretService) with .decode_identity_from_request(), encode_identity_to_token(), .validate_password(), .hash_password() , so that developer would not necessary to know how security system works to develop and use the security system. But probably I will do this later.

OK will implement this change👌

**Comment by changdiqing** [comment]
> @changdiqing are you sure that once this is merged, **everyone** could seamlessly keep their development? in another word, run those 2 servers locally will not cause any trouble.

dev DB is now set to public for every IP, I tested it on two computers with two IPs and it worked. If it works on your side, then it should work for every developer without extra configurations.

**Comment by changdiqing** [comment]
Please hold this PR for now as I still have something to double check...

**Review by Apollo1840 — COMMENTED** [review]
it looks OK for me

**Comment by Apollo1840** [comment]
> > reviewed, fantastic work!
> > I want to refactor jwt and bcrypt into one class called securityService(or secretService) with .decode_identity_from_request(), encode_identity_to_token(), .validate_password(), .hash_password() , so that developer would not necessary to know how security system works to develop and use the security system. But probably I will do this later.
> 
> OK will implement this change

Naja, maybe next time, lets do it step by step

**Comment by changdiqing** [comment]
@Apollo1840 there was one new commit from today, please review and merge if everything is OK

**Pull request merged**